simstore_and_cli/2_running_cli.md

```diff
@@ -0,0 +1,203 @@
+checking if the resulting trajectory met the minus ensemble requirements, the
+OPS CLI provides an easier approach: the `md` subcommand, which can run MD for
+either a fixed time or until a given ensemble's criteria are met.
+
+The `md` subcommand will require that we provide the ensemble name, the engine,
+and the initial frame. Since we happen to know that the frame we want is the
+last one that was saved, we can select it with `-1`. For the ensemble, we can
+see the name that was listed as missing in the error, or we can use the command
+`openpathsampling contents setup.db --table ensembles` to see all the named
+ensembles in our setup file. Noting that the ensemble name should be in quotes
+(to preserve the spaces), the command to run the MD is:
+
+```bash
+openpathsampling md setup.db -o minus-traj.db -e engine -f -1 --ensemble "condensed MIS minus"
+```
+
+We can then copy the resulting trajectory to `setup.db` with
+
+```bash
+openpathsampling append minus-traj.db -a setup.db --tag final_conditions --save-tag minus_traj
+```
+
+## Equilibrating
+
+Now we're ready to equilibrate our initial conditions before performing path
+sampling. By running `openpathsampling equlibrate --help`, we see that, in
+addition to input and output files, we need a move scheme and initial
+conditions, and we can optionally provide a "multiplier" and a number of "extra
+steps."
+
+The `equilibrate` subcommand runs until all frames from the initial conditions
+have been replaced by new, dynamically generated frames. This means that the
+unphysical high-temperature frames will be removed and replaced by frames
+generated at the correct temperature. Note that this is a bare minimum for
+reasonable initial conditions -- this does necessarily mean that the
+trajectories have sufficiently decorrelated.
+
+Additional decorrelation can be provided by the `--multiplier` and
+`--extra-steps` arguments. If that bare minimum decorrelation takes N steps,
+and the options `--multiplier 2` and `--extra-steps 10` are given, then the
+total number of path sampling steps run during equilibration will be 2*N + 10.
+
+We can select the conditions saved in tags as initial conditions to
+decorrelate. We can provide the `-t`/`--init-conds` flag more than once, and we
+will do so for the saved tags `long_traj` and `minus_traj`.
+
+Because only one move scheme is saved in `setup.db`, the `equilibrate`
+subcommand implicitly knows to use that scheme, and we don't have to specify it
+on the command line. The command to equilibrate for the minimum number of steps
+is then:
+
+```bash
+openpathsampling equilibrate setup.db -t long_traj -t minus_traj -o equil.db
+```
+
+The results of this are the initial conditions for our production run, and
+we'll copy them over to `setup.db` with the tag name `'initial_conditions'`:
+
+```bash
+openpathsampling append equil.db -a setup.db --tag final_conditions --save-tag initial_conditions
+```
```

Comment:
just FYI: this command errored out for me with:
```
openpathsampling equilibrate setup.db -t long_traj -t minus_traj -o equil.db
Traceback (most recent call last):
  File "/home/sroet/miniconda3/envs/ops/bin/openpathsampling", line 33, in <module>
    sys.exit(load_entry_point('openpathsampling-cli', 'console_scripts', 'openpathsampling')())
  File "/home/sroet/miniconda3/envs/ops/lib/python3.9/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/home/sroet/miniconda3/envs/ops/lib/python3.9/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/home/sroet/miniconda3/envs/ops/lib/python3.9/site-packages/click/core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/sroet/miniconda3/envs/ops/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/sroet/miniconda3/envs/ops/lib/python3.9/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/home/sroet/github_files/openpathsampling-cli/paths_cli/commands/equilibrate.py", line 37, in equilibrate
    init_conds=INIT_CONDS.get(storage, init_conds),
  File "/home/sroet/github_files/openpathsampling-cli/paths_cli/param_core.py", line 360, in get
    results = [super(OPSStorageLoadMultiple, self).get(storage, name)
  File "/home/sroet/github_files/openpathsampling-cli/paths_cli/param_core.py", line 360, in <listcomp>
    results = [super(OPSStorageLoadMultiple, self).get(storage, name)
  File "/home/sroet/github_files/openpathsampling-cli/paths_cli/param_core.py", line 316, in get
    raise RuntimeError(msg)
RuntimeError: Couldn't find long_traj is samplesets
```

Reply:
This should only occur if you didn't run the command:

```
openpathsampling append visit-all.db -a setup.db --tag final_conditions --save-tag long_traj
``` 

You can always check the contents of your setup file with `openpathsampling contents setup.db` -- see if there's a typo in the name of the tag (e.g., maybe you missed a character when copy-pasting that append command, and saved it with the tag `long_tra`).